pkg/controllers/deprovisioning/types.go

```diff
@@ -58,7 +59,7 @@ type Candidate struct {
 
 //nolint:gocyclo
 func NewCandidate(ctx context.Context, kubeClient client.Client, recorder events.Recorder, clk clock.Clock, node *state.StateNode,
-	nodePoolMap map[nodepoolutil.Key]*v1beta1.NodePool, nodePoolToInstanceTypesMap map[nodepoolutil.Key]map[string]*cloudprovider.InstanceType) (*Candidate, error) {
+	nodePoolMap map[nodepoolutil.Key]*v1beta1.NodePool, nodePoolToInstanceTypesMap map[nodepoolutil.Key]map[string]*cloudprovider.InstanceType, queue *orchestration.Queue) (*Candidate, error) {
```

Comment:
Should we consider factoring this in a new way as a followon? This method is falling pray to a classic limitation of static helpers -- the number of arguments keeps growing.